### PR TITLE
Improve pre-processor directives part

### PR DESCRIPTION
**Pull request by slavaleleka**
https://github.com/AdguardTeam/AdguardKnowledgeBase/issues/461

https://github.com/AdguardTeam/AdguardForiOS/issues/1104
https://github.com/AdguardTeam/AdguardForiOS/issues/819

### CODE REVIEW

File: 01.general/02.how-to-create-your-own-ad-filters/docs.en.md [hunk 1]
```diff
@@ -1801,38 +1792,104 @@ Filter URL: `https://example.org/path/filter.txt`
...
+    - `ext_ublock` — special case; this one is declared when a uBlock version of a filter is compiled by the [FiltersRegistry](https://github.com/AdguardTeam/FiltersRegistry)
+- `rules_list` — list of rules
+- `!#endif` — end of the block
 
-* If included file is not found or unavailable, the whole filter update should fail.
-* A conditional directive beginning with an `!#if` directive must explicitly be terminated with an `!#endif` directive.
-* Whitespaces matter. `!#if` is a valid directive, while `!# if` is not.
+**Examples**
+```
+! for all AdGuard propucts except of AdGuard for Safari
```
Comment (vbagirov) [inline 01.general/02.how-to-create-your-own-ad-filters/docs.en.md:1833]: except of не говорят, либо просто except, либо except for

File: 01.general/02.how-to-create-your-own-ad-filters/docs.en.md [hunk 2]
```diff
@@ -1801,38 +1792,104 @@ Filter URL: `https://example.org/path/filter.txt`
...
+!#endif
+```
 
-* Any mistake in a pre-processor directive will lead to AdGuard failing the filter update in the same way as if the filter URL was unavailable.
-* Pre-processor directives can be used in the user filter (or in the custom local filters). Same-origin limitation should be disabled for local filters.
+<a id="safari-affinity-directive"></a>
+#### Safari affinity
 
-#### What constants we declare
+Safari is notoriously known for its harsh 50k max limit for filtering rules in content blockers. But AdGuard for Safari and AdGuard for iOS max rule count raised to 300k by splitting them into several content blockers. Generally, several filters categories are more or less independent, so there is such content blockers with such categories included:
```
Comment (vbagirov) [inline 01.general/02.how-to-create-your-own-ad-filters/docs.en.md:1850]: max rule count raised -> raised max rule count

Comment (vbagirov) [reply 01.general/02.how-to-create-your-own-ad-filters/docs.en.md:1850]: Или 
But in AdGuard for Safari and AdGuard for iOS max rule count is raised to 300k by splitting them into several content blockers.

**Review by vbagirov — CHANGES_REQUESTED** [review]

File: 01.general/02.how-to-create-your-own-ad-filters/docs.ru.md [hunk 3]
```diff
@@ -1821,38 +1813,105 @@ URL фильтра: `https://example.org/path/filter.txt`
...
+! директивы также можно совмещать
+!#if (adguard_app_android)
+!#include /androidspecific.txt
+!#endif
+```
 
-* Любая ошибка в пре-процессинговой директиве приведёт к невозможности обновить фильтр, как если бы URL фильтра был недоступен.
-* Пре-процессинговые директивы могут быть использованы в Пользовательском фильтре (или в собственных локальных фильтрах). Для локальных фильтров ограничение на тот же источник не распространяется.
+<a id="safari-affinity-directive"></a>
+#### Влияние на блокировщика контента Safari
```
Comment (vbagirov) [inline 01.general/02.how-to-create-your-own-ad-filters/docs.ru.md:1870]: "Влияние на блокировщик контента Safari"

File: 01.general/02.how-to-create-your-own-ad-filters/docs.ru.md [hunk 4]
```diff
@@ -1821,38 +1813,105 @@ URL фильтра: `https://example.org/path/filter.txt`
...
 
-#### Какие константы мы объявляем
+В Safari существует строгое ограничение в 50 тысяч правил на один блокировщик контента. Но в AdGuard для Safari и в AdGuard для iOS максимальное число поддерживаемых правил увеличено до 300 тысяч. Это получилось благодаря условному их разделению на несколько блокировщиков контента. В целом, отдельные категории фильтров более-менее незавизыми, поэтому они поделены между блокировщиками контента следующим образом:
```
Comment (vbagirov) [inline 01.general/02.how-to-create-your-own-ad-filters/docs.ru.md:1872]: "независимы"

**Review by vbagirov — CHANGES_REQUESTED** [review]

**Review by vbagirov — APPROVED** [review]

**Review by ameshkov — APPROVED** [review]

**Pull request merged**